packages/actor-rdf-resolve-hypermedia-links-queue-wrapper-info-occupancy/lib/LinkQueueLogger.ts

```diff
@@ -0,0 +1,156 @@
+import { LinkQueueWrapper } from '@comunica/bus-rdf-resolve-hypermedia-links-queue';
+import type { Logger } from '@comunica/types';
+
+/**
+ * A link queue that log information about event happening in the queue
+ */
+export class LinkQueueLogger extends LinkQueueWrapper {
+  public readonly query: string;
+  private readonly logger: Logger;
+  private readonly linkProductionRatio: ILinkProductionActorRatio = {
+    pushEvents: {},
+    popEvents: {},
+  };
+
+  /**
+   *
+   * @param {ILinkQueue} linkQueue - The link queue
+   * @param {Algebra.Operation} query - The current query
+   * @param {Logger} logger - The logger where the events are output
+   */
+  public constructor(linkQueue: ILinkQueue, query: string, logger: Logger) {
+    super(linkQueue);
+    this.logger = logger;
+    this.query = query;
+  }
+
+  /**
+   * Helper function to get the reachability criteria of a link
+   * @param {ILink} link - Current link
+   * @returns {IProducedByActor | undefined } The reachability criteria with extra information about it if available
+   */
+  private static getActorProductorInformation(link: ILink): IProducedByActor | undefined {
+    const metadata = link.metadata;
+    if (metadata && metadata[PRODUCED_BY_ACTOR]) {
+      const { name, ...rest } = metadata[PRODUCED_BY_ACTOR];
+      if (name) {
+        return {
+          name,
+          metadata: Object.keys(rest).length === 0 ? undefined : rest,
+        };
+      }
+    }
+  }
+
+  /**
+   * Update the link production ratio
+   * @param {IUrlStatistic} link - current link
+   * @param {keyof ILinkProductionActorRatio} event - link queue event
+   */
+  private updateLinkProductionRatio(link: IUrlStatistic, event: keyof ILinkProductionActorRatio): void {
+    if (link.producedByActor &&
+      this.linkProductionRatio[event][link.producedByActor.name]) {
+      this.linkProductionRatio[event][link.producedByActor.name] += 1;
+    } else if (
+      this.linkProductionRatio[event][link.producedByActor?.name ?? 'unknown']) {
+      this.linkProductionRatio[event].unknown += 1;
+    } else {
+      this.linkProductionRatio[event][link.producedByActor?.name ?? 'unknown'] = 1;
+    }
+  }
+
+  /**
+   * Create the information of a link queue event
+   * @param {ILink} link - current link
+   * @param {EventType} eventType - the type of event
+   * @param {ILink|undefined} parent - the parent of the link
+   * @returns {ILinkQueueEvent} current event of the link queue
+   */
+  private createLinkQueueEvent(link: ILink, eventType: 'pushEvents' | 'popEvents', parent?: ILink): ILinkQueueEvent {
+    const linkInfo: IUrlStatistic = {
+      url: link.url,
+      producedByActor: LinkQueueLogger.getActorProductorInformation(link),
+      timestamp: performance.now(),
+      parent: parent?.url,
+    };
+    this.updateLinkProductionRatio(linkInfo, eventType);
+
+    return {
+      type: eventType.slice(0, Math.max(0, eventType.length - 1)),
```

Comment:
This is not so nice 😅 
Maybe we can just use `push` and `pop` instead of `pushEvent`/`pushEvents`, ... This would avoid the need for hacks like these.